mongodbatlas/resource_mongodbatlas_search_index.go

```diff
@@ -115,6 +119,15 @@ func returnSearchIndexSchema() map[string]*schema.Schema {
 			Type:     schema.TypeBool,
 			Optional: true,
 		},
+		"type": {
+			Type:     schema.TypeString,
+			Optional: true,
+		},
+		"fields": {
```

Comment:
have we considered adding a conflictsWith validation to ensure vectorSearch attributes are not defined together with regular search attributes?

Reply:
at the end I preferred that all those checks are done in the server. If we do them in client-side there is some risk that both logics differ now or in the future.